compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1206,11 +1208,66 @@ private boolean checkSimilarBindingPatterns(BLangBindingPattern firstBidingPatte
             case LIST_BINDING_PATTERN:
                 return checkSimilarListBindingPatterns((BLangListBindingPattern) firstBidingPattern,
                         (BLangListBindingPattern) secondBindingPattern);
+            case MAPPING_BINDING_PATTERN:
+                return checkSimilarMappingBindingPattern((BLangMappingBindingPattern) firstBidingPattern,
+                        (BLangMappingBindingPattern) secondBindingPattern);
             default:
                 return false;
         }
     }
 
+    private boolean checkSimilarMappingBindingPattern(BLangMappingBindingPattern firstMappingBindingPattern,
+                                                      BLangMappingBindingPattern secondMappingBindingPattern) {
+        if (firstMappingBindingPattern.restBindingPattern != null &&
+                secondMappingBindingPattern.restBindingPattern == null) {
+            return false;
+        }
+        if (firstMappingBindingPattern.restBindingPattern == null &&
+                secondMappingBindingPattern.restBindingPattern != null) {
+            return false;
+        }
+        List<BLangFieldBindingPattern> firstFieldBindingPatterns = firstMappingBindingPattern.fieldBindingPatterns;
+        List<BLangFieldBindingPattern> secondFieldBindingPatterns = secondMappingBindingPattern.fieldBindingPatterns;
+        if (firstMappingBindingPattern.restBindingPattern == null) {
+            if (firstFieldBindingPatterns.size() != secondFieldBindingPatterns.size()) {
+                return false;
+            }
+            return checkSimilarFieldBindingPatterns(firstFieldBindingPatterns, secondFieldBindingPatterns);
+        }
+        if (firstFieldBindingPatterns.size() > secondFieldBindingPatterns.size()) {
+            return false;
+        }
+        if (firstFieldBindingPatterns.size() == secondFieldBindingPatterns.size()) {
+            return checkSimilarFieldBindingPatterns(firstFieldBindingPatterns, secondFieldBindingPatterns);
+        }
+        return checkSimilarBindingPatterns(firstMappingBindingPattern.restBindingPattern,
+                secondMappingBindingPattern.restBindingPattern);
```

Comment:
Doesn't this always return true?

Reply:
Yes, that is correct. I did like this to improve the readability which means to indicate that finally, we should check rest-binding-pattern.